### ISSUE TITLE
Request for "IsTakingOff" or "Undocking" event trigger for shipyard blocks

### ISSUE THREAD

**Issue by CurioInventorium** [tsunamayo/Starship-EVO#4360]
### Short Description of the issue:
Sorry if this has been suggested before...
So essentially I built a large pyramid where one side opens up (reeeally slowly) to reveal the landing pad as soon as someone requests to dock.
It works really well as the doors stay open until the ship lands and due to how the logic is set up, the doors close afterwards.
Works really nicely. Only problem is now my ship is stuck inside the pyramid.

If there was a "Is Taking Off" or "Undocking" shipyard event that stays triggered while a ship that has just undocked stays within the shipyards defined zone. Then there would be a way to cause the doors to open once I've selected undock from the station menu and for them to close once the ship has left.
Hopefully I've explained what I mean.
...

### Steps to reproduce the issue:

...

### Environment
- Starship EVO version: 21w49a
- Graphics Card: TRX2060

### Additional information:

_(Screenshots, Videos, Error logs, Related issues)_
Here's some screenshots of the pyramid with doors opening and my ship landing for some flavour of what I mean.

![20211213095341_1](https://user-images.githubusercontent.com/51359034/145791878-9699a536-3a4b-4622-a86c-043c8e2fbe32.jpg)
![20211213095402_1](https://user-images.githubusercontent.com/51359034/145791880-f2685a4e-d993-4089-bb3d-92ea1d303b4a.jpg)
![20211213095437_1](https://user-images.githubusercontent.com/51359034/145791881-4f69c30a-bbac-4e60-8bdb-6974b963c0ae.jpg)
![20211213095453_1](https://user-images.githubusercontent.com/51359034/145791882-2f0c1338-c48d-4798-adbc-2161e51a6cec.jpg)
![20211213095503_1](https://user-images.githubusercontent.com/51359034/145791887-556a260d-ab12-4ed3-a986-51935400343e.jpg)
![20211213095517_1](https://user-images.githubusercontent.com/51359034/145791891-c17c2114-c80f-43af-8511-9cec3eb2047f.jpg)
![20211213095551_1](https://user-images.githubusercontent.com/51359034/145791893-22b753a9-63ac-4505-a41a-5e721de5aff2.jpg)

Please also include the `Player.log` file. You can find the log file here: `%UserProfile%\AppData\LocalLow\Moonfire Entertainment\Starship EVO\`

If you issue concerns a particular design please include any blueprint that could help to reproduce the issue.
Blueprint folder is over at:
`%UserProfile%\AppData\LocalLow\Moonfire Entertainment\Starship EVO\Save_Data\Blueprints\`
You will need to zip the folder to upload it. 
To do so Right click and choose [Send to -> compressed zip file] then drag the .zip onto your report to upload it.

Here's the pyramid.
[Dark Pyramid.zip](https://github.com/tsunamayo/Starship-EVO/files/7702833/Dark.Pyramid.zip)

**Comment by tsunamayo** [comment]
@CurioInventorium yeah it makes sense, it is a bit complex to do though, but I will have a look at it. Thanks

**Comment by CurioInventorium** [comment]
Cool. I may try experimenting with using logic to check if there's a ship docked and the player detection block (I can't remember it's proper name). That way if there's no ship docked and the player is within the bounds of the detection block, that must mean the player is in the ship and waiting to exit.
It'll break if someone just walks onto the landing pad and no ship is there. Plus I don't know how large an area the player detection block picks up.

**Comment by tsunamayo** [comment]
@CurioInventorium yeah I think the best way would be simply with the volume trigger brick, I need to make a mode for entity only. Thanks

**Comment by CurioInventorium** [comment]
@tsunamayo I did try with the volume trigger block, even though I couldn't get it large enough due to size limits.
So I set it up so that it was in the air above the landing pad where the player would be if the ship was undocked and hovering, waiting to leave.
Unfortunately the volume trigger brick seems a bit glitchy. I haven't tested properly yet, but I know it's behaviour seems to have changed since the NeoVox update.

**Comment by CurioInventorium** [comment]
I've raised issue #4435 regarding the volume trigger block not responding as expected.
